core/server/worker/src/main/java/alluxio/worker/block/SpaceReserver.java

```diff
@@ -130,6 +130,8 @@ private synchronized void reserveSpace() {
 
   @Override
   public void heartbeat() {
+    // Sync the pin list before attempting to reserve space
+    mBlockWorker.syncPinList();
```

Comment:
This might be too expensive to call on every eviction attempt. We want to keep the eviction loop fast and without dependencies on external services.

Reply:
I think there is a trade-off here for speed vs correctness.

Additionally, the PinListSync is configured by default to run every `1s` and the SpaceReserver is also configured to run every `1s`. While the two could theoretically run in parallel, I don't see an advantage to having them be independent of one another if they have the same interval configured by default.

Do you see a case where you would want these values to be different?

Reply:
The pin list being stale is still a problem even if we call it here, since the time to get the pin list can be non trivial.
An advantage to having the pin list sync run separately from the eviction process is when the pin list sync takes a non trivial amount of time to execute (ie. network, master slow). In this case eviction can continue using the stale pin list.

A stale pin list is a correctness issue which we address with the `pinOnCreate` flag (https://github.com/Alluxio/alluxio/blob/master/core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java#L305).